verifier/src/evidence.rs

```diff
@@ -113,12 +116,60 @@ mod test {
         Quote3Verifier, SignedTcbInfoVerifier, VerificationTreeDisplay, Verifier,
     };
     use alloc::format;
+    use assert_matches::assert_matches;
     use core::mem;
     use der::DateTime;
     use mc_sgx_core_sys_types::sgx_attributes_t;
     use mc_sgx_core_types::{AttributeFlags, ExtendedFeatureRequestMask};
-    use mc_sgx_dcap_sys_types::{sgx_ql_ecdsa_sig_data_t, sgx_quote3_t};
+    use mc_sgx_dcap_sys_types::{sgx_ql_ecdsa_sig_data_t, sgx_ql_qve_collateral_t, sgx_quote3_t};
     use p256::ecdsa::VerifyingKey;
+    use x509_cert::der::DecodePem;
+
+    fn collateral(tcb_info: &str) -> Collateral {
```

Comment:
Not fond of this, but I think storing the collateral in a binary format would be more unwieldy.

I'm also not sure how this will evolve as logic for verifying the cert chains starts to come in.